Update mongoose to v8

### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

Mongoose version 8 was released, this package currently only supports v6 and v7.

### Describe the solution you'd like

Bump mongoose peer dependency to allow v8.

### Teachability, documentation, adoption, migration strategy

Changelog with breaking changes from v7 to v8: https://github.com/Automattic/mongoose/blob/master/CHANGELOG.md

### What is the motivation / use case for changing the behavior?

To ensure that we are able to use the latest features and performance upgrades.

+1 when I updated to moongoose 8, these error are thrown.

npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: hms-nestjs@0.0.1
npm WARN Found: mongoose@7.6.4
npm WARN node_modules/mongoose
npm WARN   mongoose@"^8.0.0" from the root project
npm WARN   2 more (@nestjs/mongoose, @nestjs/terminus)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer mongoose@"^6.0.2 || ^7.0.0" from @nestjs/mongoose@10.0.1
npm WARN node_modules/@nestjs/mongoose
npm WARN   @nestjs/mongoose@"^10.0.1" from the root project
npm WARN   1 more (@nestjs/terminus)
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: hms-nestjs@0.0.1
npm WARN Found: mongoose@7.6.4
npm WARN node_modules/mongoose
npm WARN   mongoose@"^8.0.0" from the root project
npm WARN   2 more (@nestjs/mongoose, @nestjs/terminus)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer mongoose@"^6.0.2 || ^7.0.0" from @nestjs/mongoose@10.0.1
npm WARN node_modules/@nestjs/mongoose
npm WARN   @nestjs/mongoose@"^10.0.1" from the root project
npm WARN   1 more (@nestjs/terminus)

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -17,7 +17,7 @@ jobs:
   build:
     working_directory: ~/nest
     docker:
-      - image: cimg/node:20.3
+      - image: cimg/node:20.9
     steps:
       - checkout
       - run:

--- a/package.json
+++ b/package.json
@@ -53,7 +53,7 @@
   "peerDependencies": {
     "@nestjs/common": "^8.0.0 || ^9.0.0 || ^10.0.0",
     "@nestjs/core": "^8.0.0 || ^9.0.0 || ^10.0.0",
-    "mongoose": "^6.0.2 || ^7.0.0",
+    "mongoose": "^6.0.2 || ^7.0.0 || ^8.0.0",
     "reflect-metadata": "^0.1.12",
     "rxjs": "^7.0.0"
   },

